packages/cli/src/ConcurrencyQueue.ts

```diff
@@ -0,0 +1,30 @@
+export class ConcurrencyQueue {
+	private waiting: Array<[string, () => void]> = [];
```

Comment:
1. `readonly`
2. `waiting` is an execution status, is there a better term? `paused`? `rateLimited`? `throttled`? `onHold`?
3.  This is a nice use case for labeled tuples: `Array<[executionId: string, resolveFn: () => void]> = [];`